core/commands/ls.go

```diff
@@ -87,25 +98,31 @@ it contains, with the following format:
 			}
 		}
 
-		res.SetOutput(&LsOutput{output})
+		res.SetOutput(&LsOutput{
+			Objects: output,
+			Headers: headers,
+		})
 	},
 	Marshalers: cmds.MarshalerMap{
 		cmds.Text: func(res cmds.Response) (io.Reader, error) {
-			output := res.Output().(*LsOutput).Objects
+
+			output := res.Output().(*LsOutput)
```

Comment:
you can get the request object and the options from the response here. Not sure if thats the *right* way to do it, but its what ive been doing.